upf/test/test_upf.py

```diff
@@ -1265,6 +1274,26 @@ def forward_pdr(self, pdr_id=1):
             IE_URR_Id(id=pdr_id),
         ])
 
+    def forward_pdr2(self, pdr_id=1):
```

Comment:
It's probably better to have
`def forward_pdr(self, pdr_id=1, chid=201, ue_ip=None):`

`def ue_ip2(self):` should probably be removed, the reason for `def ue_ip(self):` is a separate impl for IPv6, and this is a concrete IPv4 test class. Just use `forward_pdr(pdr_id=3, chid=201, self.UE_IP2)` instead of `forward_pdr2(pdr_id=3)`

Reply:
`self.ue_ip_ip_address()` should be `self.ue_ip_ip_address(ue_ip=ue_ip)` in this case